Spigot-Server-Patches/0450-Add-PlayerAttackEntityCooldownResetEvent.patch

```diff
@@ -0,0 +1,37 @@
+From 39c9d80f9162c26c3a2f0bf616a39d082a05b7c9 Mon Sep 17 00:00:00 2001
+From: nossr50 <nossr50@gmail.com>
+Date: Thu, 26 Mar 2020 19:44:50 -0700
+Subject: [PATCH] Add PlayerAttackEntityCooldownResetEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityHuman.java b/src/main/java/net/minecraft/server/EntityHuman.java
+index 584166f2..3d5ddf2d 100644
+--- a/src/main/java/net/minecraft/server/EntityHuman.java
++++ b/src/main/java/net/minecraft/server/EntityHuman.java
+@@ -1,5 +1,6 @@
+ package net.minecraft.server;
+ 
++import com.destroystokyo.paper.event.player.PlayerAttackEntityCooldownResetEvent;
+ import com.google.common.collect.ImmutableMap;
+ import com.google.common.collect.Lists;
+ import com.mojang.authlib.GameProfile;
+@@ -1057,7 +1058,15 @@ public abstract class EntityHuman extends EntityLiving {
+ 
+                 f *= 0.2F + f2 * f2 * 0.8F;
+                 f1 *= f2;
+-                this.ey();
++                //Paper start - PlayerAttackEntityCooldownResetEvent
++                PlayerAttackEntityCooldownResetEvent playerAttackEntityCooldownResetEvent
++                    = new PlayerAttackEntityCooldownResetEvent((Player) this.getBukkitEntity(), entity.getBukkitEntity(), this.getCooledAttackStrength(0F));
```

Comment:
this can be a single line, space after comment start
```suggestion
+                // Paper start - PlayerAttackEntityCooldownResetEvent
+                PlayerAttackEntityCooldownResetEvent playerAttackEntityCooldownResetEvent  = new PlayerAttackEntityCooldownResetEvent((Player) this.getBukkitEntity(), entity.getBukkitEntity(), this.getCooledAttackStrength(0F));
```

Reply:
Your suggestions have been added